File: Ananya Chandra/disarium.py
#program for disarium number
st=input("enter the number \n")
l=len(st)
n=int(st)
t=n
s=0
while t>0 :
    r=t%10
    s=s+(r**l)
    l-=1
    t=t//10

if(s==n):
    print(n,"is a disarium number")
else:
    print(n,"is not a disarium number ")

File: HarshKesarwani/disarium.cpp
#include <iostream>
#include <string>
#include <cmath>
using namespace std;

int main()
{
    int num;
    cout << "Enter number to check if it is Disarium Number" << endl;
    cin >> num;
    int sum = 0;
    string copy_num = to_string(num);
    int l = copy_num.length();
    for (int x = num; x != 0; x = x / 10)
    {
        int d = x % 10;
        sum = sum + pow(d, l);
        l--;
    }
    if (sum == num)
    {
        cout << num << " IS A DISARIUM NUMBER" << endl;
    }
    else
    {
        cout << num << " IS NOT A DISARIUM NUMBER" << endl;
    }
    return 0;
}
